packages/fuels/src/cli/utils/getLatestFuelsVersion.ts

```diff
@@ -0,0 +1,32 @@
+import { checkAndLoadCache, saveToCache } from './fuelsVersionCache';
+
+export const getLatestFuelsVersion = async (): Promise<string | undefined> => {
+  const now = Date.now();
+
+  const cache = checkAndLoadCache();
+  if (cache && cache.data) {
+    return cache.data.version;
+  }
+
+  const data: { version: string } | null = await Promise.race([
+    new Promise((resolve) => {
+      setTimeout(() => resolve(null), 3000);
+    }),
+    fetch('https://registry.npmjs.org/fuels/latest').then((response) => response.json()),
+  ]);
+
+  if (!data) {
+    throw new Error('Failed to fetch latest fuels version.');
+  }
```

Comment:
As I said in [point nº 4](https://github.com/FuelLabs/fuels-ts/pull/3108#issuecomment-2346355732), nothing about this should ever break the app.

That means no throws unless you are effectively handling the exceptions.

Reply:
This is caught in the parent function that calls this function - `checkForAndDisplayUpdates`

Reply:
It won't break the app

Reply:
Oh, I missed that. Thanks.